src/extensionManager.ts

```diff
@@ -13,7 +13,11 @@ import assert from "node:assert";
  *
  * Thus, the client/server instance is effectively a singleton.
  */
+
+type NewClientCallback = (newClient: lc.LanguageClient) => void;
+
 export class CairoExtensionManager implements vscode.Disposable {
+  private newClientCallbacks: NewClientCallback[] = [];
```

Comment:
`vscode` package has a builtin `EventEmitter` class that supports resource disposition. Please use it. I know will never have to dispose anything in current state of things (and GC will handle cycle created by this callback), but good practices are good practices still

Reply:
[Use vscode's Event api for updating the client.](https://github.com/software-mansion/vscode-cairo/pull/94/commits/7a615734b022a14eef3aa3ad1c342bbd48a83485)